### PR TITLE
Move emscripten_GetProcAddress callers into C

### PR DESCRIPTION
**Pull request by kripken**
This avoids having them on the right-hand-side in deps_info, which
means that in `REVERSE_DEPS=all` mode we do not include that
function and all the GL code it calls. Instead, the callers are just more
C functions that the linker does the right thing.

### CODE REVIEW

File: system/lib/gl/libprocaddr.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  return emscripten_GetProcAddress(name);
+}
+
+void* eglGetProcAddress(const char* name) {
+  return emscripten_GetProcAddress(name);
+}
+
+void* glfwGetProcAddress(const char* name) {
+  return emscripten_GetProcAddress(name);
+}
```
Comment (sbc100) [inline system/lib/gl/libprocaddr.c:22]: Would it not make sense to these to live in their various different libraries?  

Is `emscripten_GetProcAddress` just for GL symbols?   I wonder if it should be renamed to reflect that?

**Review by sbc100 — COMMENTED** [review]
Awesome!!

File: system/lib/gl/libprocaddr.c (hunk 1 above)
Comment (kripken) [reply system/lib/gl/libprocaddr.c:22]: We don't have compiled libraries for them, and adding new ones just for what are basically aliases seemed like overkill Perhaps we can even just make them aliases in the headers, if you prefer that?

Yes, that function is just for GL symbols. I'm not opposed to renaming it (separate from this PR I assume?)

**Comment by sbc100** [comment]
Can we also do the same for `emscripten_GetAlProcAddress` callers?

When those entries removed too the vast majority of the effect of `REVERSE_DEPS=auto` is just the size of dlmalloc/dlfree.

```
-rw-r----- 1 sbc primarygroup 123323 Feb 23 19:36 all.js
-rwxr-x--- 1 sbc primarygroup  14801 Feb 23 19:36 all.wasm
-rw-r----- 1 sbc primarygroup 120058 Feb 23 19:36 auto.js
-rwxr-x--- 1 sbc primarygroup    790 Feb 23 19:36 auto.wasm
```

JS size hardly effected and wasm size goes up by ~10k which is basically the cost of malloc/free:

```
 - func[29] size=6095 <dlmalloc>
 - func[30] size=1732 <dlfree>
 - func[31] size=134 <dlrealloc>
 - func[32] size=1015 <try_realloc_chunk>
 - func[33] size=23 <dlmemalign>
 - func[34] size=412 <internal_memalign>
 - func[35] size=1661 <dispose_chunk>
```

**Comment by kripken** [comment]
@sbc100 Yes, I think those can be done as well. Do you prefer them in this PR or in another?

But ping on my questions from earlier, this PR is waiting on a response from you there I believe? (https://github.com/emscripten-core/emscripten/pull/13524#discussion_r578014928)

File: system/lib/gl/libprocaddr.c (hunk 1 above)
Comment (sbc100) [reply system/lib/gl/libprocaddr.c:22]: I'm find with this now if tests pass.

**Comment by sbc100** [comment]
Followup PR is fine.

**Comment by kripken** [comment]
Looks like tests failed due to SDL2 linking in its own `SDL_GL_GetProcAddress`. I made the symbol weak to avoid that - thoughts?

**Comment by sbc100** [comment]
Making it weak sounds fine.  Another alternative might be to put each function in its own object file.. but that might cause complications with `--whole-archive` users.

**Pull request merged**